pytroll_collectors/s3stalker.py

```diff
@@ -23,26 +19,125 @@
 
 logger = logging.getLogger(__name__)
 
 
+class S3StalkerRunner(Thread):
+    """Runner for stalking for new files in an S3 object store."""
+
+    def __init__(self, bucket, config, startup_timedelta_seconds):
+        """Initialize the S3Stalker runner class."""
+        super().__init__()
+
+        self.bucket = bucket
+        self.config = config
+        self.startup_timedelta_seconds = startup_timedelta_seconds
+        self.time_back = self.config['timedelta']
+        self.publisher = None
+        self.loop = False
+        self._set_signal_shutdown()
+
+    def _set_signal_shutdown(self):
+        """Set a signal to handle shutdown."""
+        signal.signal(signal.SIGTERM, self.signal_shutdown)
+
+    def _setup_and_start_communication(self):
+        """Set up the Posttroll communication and start the publisher."""
+        logger.debug("Starting up... ")
+        self.publisher = create_publisher_from_dict_config(self.config['publisher'])
+        self.publisher.start()
+        self.loop = True
+
+    def signal_shutdown(self, *args, **kwargs):
+        """Shutdown the S3 Stalker daemon/runner."""
+        self.close()
+
+    def run(self):
+        """Start the s3-stalker daemon/runner in a thread."""
+        self._setup_and_start_communication()
+
+        waiting_time = timedelta(**self.time_back)
+        wait_seconds = waiting_time.total_seconds()
+
+        first_run = True
+        config = self.config
+        last_fetch_time = None
+        while self.loop:
+            last_fetch_time = self.do_fetch_most_recent(last_fetch_time, first_run, config)
+            first_run = False
+            logger.debug("Waiting %d seconds", wait_seconds)
+            time.sleep(max(wait_seconds, 0))
+
+    def do_fetch_most_recent(self, last_fetch_time, first_run, config):
+        """Do a fecth of the most recent files and create and send the messages."""
+        config['timedelta'] = self._get_timedelta(last_fetch_time, is_first_run=first_run)
+
+        last_fetch_time = get_last_fetch()
+        logger.info("Last fetch time...: %s", str(last_fetch_time))
+
+        self._process_messages(config)
+        return last_fetch_time
+
+    def _process_messages(self, config):
+        """Go through all messages in list and publish them one after the other."""
+        messages = create_messages_for_recent_files(self.bucket, config)
+        for message in messages:
+            logger.info("Publishing %s", str(message))
+            self.publisher.send(str(message))
+
+    def _get_timedelta(self, last_fetch_time, is_first_run):
+        """Get the seconds for the time window to search for (new) files."""
+        if is_first_run:
+            logger.info('Create messages with urls for most recent files only')
+            logger.info('On start up we consider files with age up to %d seconds from now',
+                        self.startup_timedelta_seconds)
+            return {'seconds': self.startup_timedelta_seconds}
```

Comment:
The `datetime.timedelta()` function is only given seconds, so to simplify just return the value without a dictionary, and use `timedelta(seconds=return_value_from_here)`.